libs/vkd3d/command.c

```diff
@@ -3372,26 +3372,37 @@ static void d3d12_command_list_update_root_descriptors(struct d3d12_command_list
     const struct d3d12_root_parameter *root_parameter;
     VkDescriptorSet descriptor_set = VK_NULL_HANDLE;
     union root_parameter_data root_parameter_data;
-    unsigned int root_parameter_index, va_count;
     unsigned int descriptor_write_count = 0;
+    unsigned int root_parameter_index;
+    unsigned int va_count = 0;
+    uint64_t dirty_push_mask;
 
     if (root_signature->flags & VKD3D_ROOT_SIGNATURE_USE_ROOT_DESCRIPTOR_SET)
     {
         /* Ensure that we populate all descriptors if push descriptors cannot be used */
-        bindings->root_descriptor_dirty_mask |= bindings->root_descriptor_active_mask & root_signature->root_descriptor_mask;
+        bindings->root_descriptor_dirty_mask |=
+                bindings->root_descriptor_active_mask &
+                (root_signature->root_descriptor_raw_va_mask | root_signature->root_descriptor_push_mask);
 
         descriptor_set = d3d12_command_allocator_allocate_descriptor_set(
                 list->allocator, root_signature->vk_root_descriptor_layout, VKD3D_DESCRIPTOR_POOL_TYPE_STATIC);
     }
 
-    if (!(list->device->bindless_state.flags & VKD3D_RAW_VA_ROOT_DESCRIPTOR))
+    if (bindings->root_descriptor_dirty_mask)
     {
-        /* TODO bind null descriptors for inactive root descriptors */
-        bindings->root_descriptor_dirty_mask &= bindings->root_descriptor_active_mask;
+        /* If any push descriptor is dirty, we need to update all of them. */
```

Comment:
Wording seems misleading here since this is about the raw VA ones, not the actual push descriptors.

Reply:
Right.

Reply:
Fixed.